sdk/go/getPublicKey.go

```diff
@@ -0,0 +1,101 @@
+package ehsm
+
+import (
+    "crypto"
+    hmac "crypto/hmac"
+    sha256 "crypto/sha256"
+    "encoding/base64"
+    "encoding/json"
+    "fmt"
+    ioutil "io/ioutil"
+    "time"
+    "strconv"
+    "errors"
+    "crypto/x509"
+    "encoding/pem"
+
+    "github.com/iancoleman/orderedmap"
+)
+
+type PubkeyResponse struct {
+    Code int `json:"code"`
+    Result struct {
+        Keyid string `json:"keyid"`
+        Pubkey string `json:"pubkey"`
+    } `json:"result"`
+}
+
+type PEMType string
+
+const (
+    // PublicKeyPEMType is the string "PUBLIC KEY" to be used during PEM encoding and decoding
+    PublicKeyPEMType PEMType = "PUBLIC KEY"
+    // PKCS1PublicKeyPEMType is the string "RSA PUBLIC KEY" used to parse PKCS#1-encoded public keys
+    PKCS1PublicKeyPEMType PEMType = "RSA PUBLIC KEY"
+)
+
+func (c *Client) Getpubkey(keyid string) (crypto.PublicKey, error) {
+    // make JSON for getpubkey
+    payload := orderedmap.New()
+    if keyid != ""{
+        payload.Set("keyid", keyid)
+    }
```

Comment:
same. why not return error directly?

Reply:
if (keyid == "")  {
   return "", fmt.Errorf("Please input keyid.")
}
payload.Set("keyid", keyid)